src/metabase/lib/schema/validate.cljc

```diff
@@ -0,0 +1,38 @@
+(ns metabase.lib.schema.validate
+  (:require
+   [metabase.lib.schema.common :as common]
+   [metabase.util.malli.registry :as mr]))
+
+(mr/def ::missing-column
+  [:map
+   [:type [:= :validate/missing-column]]
+   [:name :string]])
+
+(mr/def ::missing-table-alias
+  [:map
+   [:type [:= :validate/missing-table-alias]]
+   [:name :string]])
+
+(mr/def ::duplicate-column
+  [:map
+   [:type [:= :validate/duplicate-column]]
+   [:name :string]])
+
+(mr/def ::syntax-error
+  [:map
+   [:type [:= :validate/syntax-error]]])
+
+(mr/def ::validation-error
+  [:map
+   [:type [:= :validate/validation-error]]
+   [:message :string]])
+
+(mr/def ::error
+  [:and
+   [:map [:type {:decode/normalize common/normalize-keyword} :keyword]]
```

Comment:
maybe worth saying `:type` has to be `qualified-keyword?` instead of `:keyword`, or better yet also saying 

```clj
[:fn 
 {...}
 #(= (namespace %) "validate")] 
```

to make the assumption/expectation that all validation errors have the `validate` namespace concrete rather than implied